CGColorSpace should be serialized without using an opaque Dictionary

gavin-apple opened the pull request:
#### 5e417d276406b07b6b6dff91473aed8570762e97
<pre>
CGColorSpace should be serialized without using an opaque Dictionary
<a href="https://bugs.webkit.org/show_bug.cgi?id=302324">https://bugs.webkit.org/show_bug.cgi?id=302324</a>
<a href="https://rdar.apple.com/164472019">rdar://164472019</a>

Reviewed by Pascoe.

Breaks CGColorSpace down into basic CoreIPC primitives as opposed
to serializing as an opaque Dictionary.

* LayoutTests/ipc/decode-destinationColorSpace-null-crash-expected.txt: Removed.
* LayoutTests/ipc/decode-destinationColorSpace-null-crash.html: Removed.
* LayoutTests/ipc/decode-feConvolveMatrix-kernelSize-overflow.html:
* LayoutTests/ipc/empty-svgfilterrenderer-expression-crash.html:
* LayoutTests/ipc/serialized-type-info.html:
* Source/WebKit/Scripts/webkit/opaque_ipc_types.tracking.in:
* Source/WebKit/Shared/cf/CoreIPCCGColorSpace.h:
(WebKit::CoreIPCCGColorSpace::CoreIPCCGColorSpace):
(WebKit::CoreIPCCGColorSpace::toCF const):
* Source/WebKit/Shared/cf/CoreIPCCGColorSpace.mm: Added.
(WebKit::CoreIPCCGColorSpace::serializableColorSpace):
(WebKit::CoreIPCCGColorSpace::CoreIPCCGColorSpace):
(WebKit::CoreIPCCGColorSpace::toCF const):
* Source/WebKit/Shared/cf/CoreIPCCGColorSpace.serialization.in:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/303497@main">https://commits.webkit.org/303497@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/18d37fcc5a223bce1abcf13f8a3785c513c471a1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/132607 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/5102 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/43692 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/140128 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/84625 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/66ee0d1b-eb46-429f-9f1e-6248d62fe269) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/5303 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/4861 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/101375 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/68672 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/1230bc4f-3358-4c63-aaa1-894cddf46f06) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/135553 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/3730 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/118807 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/82172 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/8aa8d977-5d04-43df-835b-f33c2a6c5458) 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/131956 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/3618 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/1377 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/83363 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/112545 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/36922 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/142783 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/4772 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/37511 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/109752 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/4854 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/4123 "Found 1 new test failure: imported/w3c/web-platform-tests/fullscreen/rendering/fullscreen-root-block-scroll.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109933 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27862 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/3632 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/115079 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/58212 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/4826 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/33427 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/4662 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/68277 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/4917 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/4783 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

gavin-apple commented on Source/WebKit/Shared/cf/CoreIPCCGColorSpace.h:
```diff
@@ -27,56 +27,55 @@
 }
 
-#endif
+#endif // PLATFORM(CF)
```
This is the wrong macro name. Remove.

pascoej commented on Source/WebKit/Shared/cf/CoreIPCCGColorSpace.mm:
This is same condition as 54-55.

pascoej reviewed (changes requested):
nit: commit message s/oposed/opposed/

gavin-apple committed:
CGColorSpace should be serialized without using an opaque Dictionary
https://bugs.webkit.org/show_bug.cgi?id=302324
rdar://164472019

Reviewed by Pascoe.

Breaks CGColorSpace down into basic CoreIPC primitives as opposed
to serializing as an opaque Dictionary.

* LayoutTests/ipc/decode-destinationColorSpace-null-crash-expected.txt: Removed.
* LayoutTests/ipc/decode-destinationColorSpace-null-crash.html: Removed.
* LayoutTests/ipc/decode-feConvolveMatrix-kernelSize-overflow.html:
* LayoutTests/ipc/empty-svgfilterrenderer-expression-crash.html:
* LayoutTests/ipc/serialized-type-info.html:
* Source/WebKit/Scripts/webkit/opaque_ipc_types.tracking.in:
* Source/WebKit/Shared/cf/CoreIPCCGColorSpace.h:
(WebKit::CoreIPCCGColorSpace::CoreIPCCGColorSpace):
(WebKit::CoreIPCCGColorSpace::toCF const):
* Source/WebKit/Shared/cf/CoreIPCCGColorSpace.mm: Added.
(WebKit::CoreIPCCGColorSpace::serializableColorSpace):
(WebKit::CoreIPCCGColorSpace::CoreIPCCGColorSpace):
(WebKit::CoreIPCCGColorSpace::toCF const):
* Source/WebKit/Shared/cf/CoreIPCCGColorSpace.serialization.in:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/303497@main
